static/js/components/DiskDeviceSelector.tsx

```diff
@@ -0,0 +1,94 @@
+import React, {
+  ChangeEvent,
+  Dispatch,
+  FC,
+  SetStateAction,
+  useState,
+} from "react";
+import { Input, Select } from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import { NotificationHelper } from "../types/notification";
+import { queryKeys } from "../util/queryKeys";
+import { LxdDiskDevice } from "../types/device";
+import { fetchStorages } from "../api/storages";
+
+interface Props {
+  notify: NotificationHelper;
+  setDiskDevice: Dispatch<SetStateAction<LxdDiskDevice | null>>;
+}
+
+const DiskDeviceSelector: FC<Props> = ({ notify, setDiskDevice }) => {
+  const [pool, setPool] = useState<string>("");
+  const [path, setPath] = useState<string>("");
```

Comment:
Just like the LimitSelector, this component can be stateless, when the parent passes in the diskDevice as a prop.